src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String EXCEPTION_TEXT = "Index out of bound exception.";
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            Node<T> first = new Node<>(null, value, null);
+            head = first;
+            tail = first;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(EXCEPTION_TEXT);
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> current = getNode(index);
+            Node<T> left = current.prev;
+            Node<T> newNode = new Node<>(left, value, current);
+            if (left == null) {
+                head = newNode;
+            } else {
+                current.prev = newNode;
+                left.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexChecker(index);
```

Comment:
```suggestion
        checkIndex(index);
```
use verbs for method names